pkg/infoutil/infoutil_linux.go

```diff
@@ -132,3 +136,17 @@ func fulfillPlatformInfo(info *dockercompat.Info) {
 		info.MemTotal = memLimit.MemTotal
 	}
 }
+func checkSupportBlockIOWeight() bool {
+	if CgroupsVersion() == "1" {
+		blkioController := "/sys/fs/cgroup/blkio"
+		blkioWeightPath := filepath.Join(blkioController, "blkio.weight")
+		if _, err := os.Stat(blkioWeightPath); errors.Is(err, fs.ErrNotExist) {
+			// if bfq io scheduler is used, the blkio.weight knob will be exposed as blkio.bfq.weight
+			blkioBfqWeightPath := filepath.Join(blkioController, "blkio.bfq.weight")
+			if _, err := os.Stat(blkioBfqWeightPath); errors.Is(err, fs.ErrNotExist) {
+				return false
+			}
+		}
+	}
+	return true
```

Comment:
You can use https://github.com/moby/moby/blob/d7f2e472244ae1043182809779170b50d830c90a/pkg/sysinfo/sysinfo.go#L18

This also supports cgroup v2.

Reply:
(See `fulfillPlatformInfo()` for the usage of the moby sysinfo)